src/terralib.lua

```diff
@@ -555,13 +555,13 @@ function T.terrafunction:setinlined(v)
     assert(self:isdefined(), "attempting to set the inlining state of an undefined function")
     self.definition.alwaysinline = not not v
     assert(not (self.definition.alwaysinline and self.definition.dontoptimize),
-           "setinlined(true) and setoptimized(false) are incompatible")
+            "setinlined(true) and setoptimized(false) are incompatible")
```

Comment:
I don't begrudge removing the trailing whitespace in most of this file, but this change seems to not follow the rest of the formatting of the code generally.

Is there a standardized code formatter you used to generate this formatting? We should probably talk about this.

I will say that the whitespace formatting changes do make the diffs rather hard to read, simply due to sheer volume. I'm not sure whether they can be easily backed out or applied separately. I suppose one option is to apply the format to the master branch first and rebase/merge this branch on top.

Reply:
I don't know exactly how that happened. I am using CLion for the first time, because it has debug capability for the C++ as well as the Lua side of the compiler stack. Maybe I pushed a wrong button. 

The implementation has gone through quite some addition and shaving. Maybe it's good I close and reopen the pull request such that we get a clean commit history. That also gives me the change to review all the parts again myself and add some additional documentation here and there.